store/src/commonMain/kotlin/com/dropbox/external/store5/impl/ShareableMarket.kt

```diff
@@ -0,0 +1,431 @@
+        writeRequests.access {
+
+            for (writeRequest in writeRequestQueue) {
+                if (writeRequest.created <= created) {
+                    val output = converter(input)
+                    val fetchResponse = RemoteResult.Success(output)
+                    (writeRequest.onCompletion as OnRemoteCompletion<Output>).onSuccess(fetchResponse)
+                } else {
+                    outstandingWriteRequests.add(writeRequest)
+                }
+            }
+            resetWriteRequestQueue(key, outstandingWriteRequests)
+        }
+    }
+
+    private suspend fun <Input : Any, Output : Any> tryUpdateServer(writer: Writer<Key, Input, Output>): Boolean {
+        return when (val result = tryPost<Input, Output>(writer.key)) {
+            is RemoteResult.Success -> {
+                updateWriteRequestQueue(
+                    key = writer.key,
+                    input = result.value,
+                    created = writer.updater.created,
+                    converter = writer.updater.converter
+                )
+
+                val output = writer.updater.converter(result.value)
+                val marketResponse = MarketResponse.Success(output, origin = MarketResponse.Companion.Origin.Remote)
+
+                conflictResolver.deleteFailedWriteRecord(writer.key)
+                writer.onCompletions.forEach { onCompletion -> onCompletion.onSuccess(marketResponse) }
+                true
+            }
+
+            is RemoteResult.Failure -> {
+                conflictResolver.setLastFailedWriteTime(writer.key, Clock.System.now().epochSeconds)
+                false
+            }
+        }
+    }
+
+    override suspend fun <Input : Any, Output : Any> write(writer: Writer<Key, Input, Output>): Boolean {
+        val broadcast = requireBroadcast<Output>(writer.key)
+
+        val responseLocalWrite =
+            MarketResponse.Success(writer.input as Output, origin = MarketResponse.Companion.Origin.LocalWrite)
+        broadcast.emit(responseLocalWrite)
+        stores.forEach { store -> (store.write as Write<Key, Input>).invoke(writer.key, writer.input) }
+
+        addOrInitWriteRequest(writer.key, writer.updater)
+
+        return tryUpdateServer(writer)
+    }
+
+    private fun <Output : Any> lastOrNull(key: Key): Output? {
+        val last = getBroadcast<Output>(key)?.replayCache?.last()
+        if (last is MarketResponse.Success) {
+            return last.value
+        }
+        return null
+    }
+
+    private suspend fun <Output : Any> load(key: Key) {
+        val broadcast = requireBroadcast<Output>(key)
+
+        for (store in stores) {
+            try {
+                val last = (store.read(key) as Flow<Output?>).lastOrNull()
+                if (last != null) {
+                    broadcast.emit(MarketResponse.Success(last, origin = MarketResponse.Companion.Origin.Store))
+                    return
+                }
+            } catch (_: Throwable) {
+            }
```

Comment:
At least a noop comment would be nice

Reply:
https://github.com/MobileNativeFoundation/Store/pull/450/commits/ffdc62310029c0749c1cfe967a3a4d89c5cc4e67